src/renderers/common/XRManager.js

```diff
@@ -788,44 +792,71 @@ class XRManager extends EventDispatcher {
 
 		const translationObject = new Vector3();
 		const quaternionObject = new Quaternion();
+		const renderer = this._renderer;
 
 		const wasPresenting = this.isPresenting;
-		const rendererOutputTarget = this._renderer.getOutputRenderTarget();
-		const rendererFramebufferTarget = this._renderer._frameBufferTarget;
+		const rendererOutputTarget = renderer.getOutputRenderTarget();
+		const rendererFramebufferTarget = renderer._frameBufferTarget;
 		this.isPresenting = false;
 
+		const rendererSize = new Vector2();
+		renderer.getSize( rendererSize );
+		const rendererQuad = renderer._quad;
+
 		for ( const layer of this._layers ) {
 
 			layer.renderTarget.isXRRenderTarget = this._session !== null;
 			layer.renderTarget.hasExternalTextures = layer.renderTarget.isXRRenderTarget;
+			renderer.setSize( layer.renderTarget.width, layer.renderTarget.height, false, false );
```

Comment:
I think you can move this line in the below `if` block. `setSize()` only matters if rendering into the default framebuffer. When rendering into a render target defined via  `setRenderTarget()`, the dimensions from the render target are used.

Reply:
I wonder if we should change the renderer so `setOutputRenderTarget()` behaves like `setRenderTarget()`. Meaning, when an output render target is defined, the dimensions of the internal framebuffer target is derived from it. Right now, we only do this for `depth`:

https://github.com/mrdoob/three.js/blob/2d978cb7ee65e4882ddfce48b761eb43893072fc/src/renderers/common/Renderer.js#L1246

However, keeping the dimensions of the output render target in sync with the internal framebuffer target could simplify things.

 We use `setOutputRenderTarget()` only in context of XR right now (to define a different output target than the default framebuffer) so if you think that design would be helpful we maybe can give it a shot.

Reply:
> I wonder if we should change the renderer so `setOutputRenderTarget()` behaves like `setRenderTarget()`. Meaning, when an output render target is defined, the dimensions of the internal framebuffer target is derived from it. 

Maybe we can try that in a follow up PR? I'm a bit worried that some code might look at the renderer's size so if we don't update it, they might make the wrong decisions.